platforms/core-configuration/kotlin-dsl-integ-tests/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/PrecompiledScriptPluginAccessorsIntegrationTest.kt

```diff
@@ -363,7 +363,7 @@ class PrecompiledScriptPluginAccessorsIntegrationTest : AbstractKotlinIntegratio
                 attributes {
                     attribute(
                         LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE,
-                        objects.named(LibraryElements.JAR)
+                        named(org.gradle.api.attributes.LibraryElements::class, LibraryElements.JAR)
```

Comment:
Thats an interesting test: the member function `named` of `ConfigurationContainer` takes precedence over the generated Kotlin extension function `inline fun<reified T: Named> named(name: kotlin.String): T`, because members always win, unless there is a (non-existing) Kotlin DSL marker annotation.

Reply:
You mentioned the kotlin DSL marker. Is there something we can add to the generated extension to make it take precedence over the named method on ConfigurationContainer?

Reply:
I did some digging how the `reified fun<T> ObjectFactory.named` method is implemented. I thought it is somehow magically generated, but it is just [manually hand written](https://github.com/gradle/gradle/blob/cb0c615fb8e3690971bb7f89ad80f58943360624/platforms/core-configuration/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/ObjectFactoryExtensions.kt#L36).

So I tried the same for `AttributeContainer` and it does work! Will create a follow-up PR in the next minutes

Reply:
Created #35246